core/src/main/java/com/g16/feyrune/model/overworld/map/MapParser.java

```diff
@@ -37,14 +37,37 @@ public static Map parseMapFile(String filePath) {
 
         Iterable<Point> encounterTilePositions = parseEncounterTiles(doc, mapSize);
 
+        String terrainType = parseTerrainType(doc);
```

Comment:
Terraint type should probably be an enum, and we throw an error if is not a valid terrain.

Reply:
I dont think its necessary to use an enum to store the terraindata, strings are more versatile and we can do the same check if its valid with strings as well.